lib/src/iterable_extensions.dart

```diff
@@ -582,6 +582,59 @@ extension IterableExtension<T> on Iterable<T> {
     return true;
   }
 
+  /// Whether no element and index satisfies [test].
+  ///
+  /// Returns true if no element and index satisfies [test],
+  /// and false if at least one does.
+  ///
+  /// Equivalent to `iterable.everyIndexed((i, x) => !test(i, x))` or
+  /// `!iterable.anyIndexed(test)`.
+  bool noneIndexed(bool Function(int, T) test) {
+    var index = 0;
+    for (var element in this) {
+      if (test(index++, element)) return false;
+    }
+    return true;
+  }
```

Comment:
I'd use `iterable.indexed.none((p) => test(p.$1, p.$2))` and wait for pattern parameters to make that just 
```dart
iterable.indexed.none((var (index, element)) { 
  // test here
});
```
I don't think we should be adding methods that introduce indexes, instead we should make things work with the `.indexed`  we have in the platform libraries, which works on all iterables.

Reply:
That's a good point - `indexed` is pretty composable with most of the operations other than the various `*WhereIndexed*`. We might even consider deprecating them and add a fix to use the `.indexed.operation` variant.

Reply:
> [..] wait for pattern parameters to make that just
> 
> ```dart
> iterable.indexed.every((var (index, element)) { 
>   // test here, negated.
> });
> ```

This doesn't seem to work, is it planed for future or the syntax is different?

<img width="953" alt="image" src="https://github.com/dart-lang/collection/assets/833473/f5e3a678-862e-4845-afef-7f8f24eee0b5">

Reply:
Unrelated to the topic but spotted a compiler crash with it also btw https://github.com/dart-lang/sdk/issues/53268

Reply:
I removed all uses of the .thingsIndexed in my app which made things more fun (as I can use comprehension also better), just that I wished I had access to the mentioned destructing in argument syntax which I guess in future.

Reply:
> is it planed for future or the syntax is different?

It's something we're considering for the future, but we do not have concrete plans. https://github.com/dart-lang/language/issues/3001

Reply:
> > is it planed for future or the syntax is different?
> 
> It's something we're considering for the future, but we do not have concrete plans. [dart-lang/language#3001](https://github.com/dart-lang/language/issues/3001)

It's totally understandable 🙏